### PR TITLE
py-transformers: Update to version latest version

### PR DESCRIPTION
**Pull request by essandess**
py-transformers: Update to version 3.3.1, Submission of dependencies

Note: https://github.com/macports/macports-ports/pull/7791 is also a dependency

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H2
Xcode 12.0.1 12A7300

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-datasets/Portfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-datasets
+version             1.1.2
```
Comment (reneeotten) [inline python/py-datasets/Portfile:7]: please update to latest version (1.1.3)

File: python/py-datasets/Portfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    checksums       rmd160  a31d1a54e4bc6c2136115157a26589e7f2a4e5b0 \
+                    sha256  265e007e315dcd3ef669fd56301ab0c6e863606c54dc4d37c2173bd05eba5efa \
+                    size    124943
+
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_run-append \
+                    port:py${python.version}-dill \
+                    port:py${python.version}-filelock \
```
Comment (reneeotten) [inline python/py-datasets/Portfile:33]: this is no longer a dependency

File: python/py-fire/Portfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-fire
+version             0.3.1
+platforms           darwin
+license             Apache
```
Comment (reneeotten) [inline python/py-fire/Portfile:9]: this should be `Apache-2`

File: python/py-datasets/Portfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-datasets
+version             1.1.2
+platforms           darwin
```
Comment (reneeotten) [inline python/py-datasets/Portfile:8]: it this port doesn't compile anything (please check!), then add `supported_archs noarch`

File: python/py-fire/Portfile [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-fire
+version             0.3.1
+platforms           darwin
```
Comment (reneeotten) [inline python/py-fire/Portfile:8]: if this port doesn't compile anything (please check!), then add supported_archs noarch

File: python/py-multiprocess/Portfile [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-multiprocess
+version             0.70.10
+platforms           darwin
```
Comment (reneeotten) [inline python/py-multiprocess/Portfile:8]: if this port doesn't compile anything (please check!), then add supported_archs noarch

File: python/py-semantic_version/Portfile [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-semantic_version
+version             2.8.5
+platforms           darwin
```
Comment (reneeotten) [inline python/py-semantic_version/Portfile:8]: if this port doesn't compile anything (please check!), then add supported_archs noarch

File: python/py-setuptools-rust/Portfile [hunk 8]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-setuptools-rust
+version             0.11.3
+platforms           darwin
```
Comment (reneeotten) [inline python/py-setuptools-rust/Portfile:8]: if this port doesn't compile anything (please check!), then add supported_archs noarch

File: python/py-timeout-decorator/Portfile [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-timeout-decorator
+version             0.4.1
+platforms           darwin
```
Comment (reneeotten) [inline python/py-timeout-decorator/Portfile:8]: if this port doesn't compile anything (please check!), then add supported_archs noarch

File: python/py-tokenizers/Portfile [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+homepage            https://github.com/huggingface/tokenizers
+
+python.versions     37 38
+
+if {${name} ne ${subport}} {
+    checksums       ${distname}${extract.suffix} \
+                    rmd160  103bd238583abb0abb2a867ea59eef2c134e7c75 \
+                    sha256  0bfa960345e114efd553e265e32eca6d79861abe24925fba903c925f8760f795 \
+                    size    170434
```
Comment (reneeotten) [inline python/py-tokenizers/Portfile:25]: please move the checksums outside of the `if`-block; also `${distname}${extract.suffix}` shouldn't be needed I think

File: python/py-tokenizers/Portfile [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+    depends_build-append \
+                    port:py${python.version}-setuptools-rust \
+                    port:py${python.version}-setuptools_scm
+
+    build.cmd       ${python.bin} setup.py --no-user-cfg build
```
Comment (reneeotten) [inline python/py-tokenizers/Portfile:31]: why are you overriding the `build.cmd` - why does the default not work?

File: python/py-xxhash/Portfile [hunk 12]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-xxhash
+version             2.0.0
+platforms           darwin
```
Comment (reneeotten) [inline python/py-xxhash/Portfile:8]: if no code is compiled, please add `supported_archs noarch`

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks @essandess, if you can take care of these comments and if needed update to the latest versions ; we should be able to get this merged soon-ish

File: python/py-xxhash/Portfile (hunk 12 above)
Comment (essandess) [reply python/py-xxhash/Portfile:8]: Code is compiled.

File: python/py-setuptools-rust/Portfile (hunk 8 above)
Comment (essandess) [reply python/py-setuptools-rust/Portfile:8]: Thanks done.

File: python/py-tokenizers/Portfile (hunk 10 above)
Comment (essandess) [reply python/py-tokenizers/Portfile:25]: Thanks done.

File: python/py-tokenizers/Portfile (hunk 11 above)
Comment (essandess) [reply python/py-tokenizers/Portfile:31]: Thanks done. IIRC this was necessary in a previous version for some reason, bot no longer so.

File: python/py-timeout-decorator/Portfile (hunk 9 above)
Comment (essandess) [reply python/py-timeout-decorator/Portfile:8]: Thanks done.

File: python/py-semantic_version/Portfile (hunk 7 above)
Comment (essandess) [reply python/py-semantic_version/Portfile:8]: Thanks done.

File: python/py-multiprocess/Portfile (hunk 6 above)
Comment (essandess) [reply python/py-multiprocess/Portfile:8]: Thanks done.

File: python/py-fire/Portfile (hunk 5 above)
Comment (essandess) [reply python/py-fire/Portfile:8]: Thanks done.

File: python/py-fire/Portfile (hunk 3 above)
Comment (essandess) [reply python/py-fire/Portfile:9]: Thanks done.

File: python/py-datasets/Portfile (hunk 4 above)
Comment (essandess) [reply python/py-datasets/Portfile:8]: Thanks done.

File: python/py-datasets/Portfile (hunk 2 above)
Comment (essandess) [reply python/py-datasets/Portfile:33]: Thanks done.

File: python/py-datasets/Portfile (hunk 1 above)
Comment (essandess) [reply python/py-datasets/Portfile:7]: Thanks done.

**Comment by reneeotten** [comment]
there seems to be a missing dependency:
```
port lint --nitpick py38-datasets:
Error: Unknown dependency: py38-pyarrow
```

perhaps the port you want here is `py-arrow` instead (that is available, but I haven't checked if it's the correct one...)

**Comment by essandess** [comment]
There is a dependency hell issue with Apache Arrow here—we [want](https://github.com/huggingface/datasets/blob/78538c7597ca4e2b87ab4bda2dbbc7a3c08c1d2d/setup.py#L68) `pyarrow`. The ports `py-pyarrow` (and `py38-pyarrow`) are submitted in https://github.com/macports/macports-ports/pull/7791. 

But this requires the `grpc` bug fix also included in https://github.com/macports/macports-ports/pull/7791.

The version of `grpc` in that PR (as of now) is itself outdated, but the latest two releases of `grpc` don't build on macOS; see https://github.com/grpc/grpc/issues/24412 and https://github.com/grpc/grpc/issues/24902.

I have a submission to update to the latest `grpc` ready to go that also integrates `py-grpcio` and `py-grpcio-tools` into a single Portfile, but Google must first fix these issues.

Then we can see if this version works with Apache Arrow and merge https://github.com/macports/macports-ports/pull/7791, then we can merge this PR.

> there seems to be a missing dependency:
> 
> ```
> port lint --nitpick py38-datasets:
> Error: Unknown dependency: py38-pyarrow
> ```
> 
> perhaps the port you want here is `py-arrow` instead (that is available, but I haven't checked if it's the correct one...)

**Comment by essandess** [comment]
@reneeotten 

All dependencies of this PR have now been merged, and I confirm it builds, installs, and runs on 10.15 and 11.

I believe that the CI errors are simply timeouts—this uses both tensorflow and torch. The CI build works for Python 39, but then times out while building Python 38.

Would you please merge this update?

**Pull request merged**